kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -63,9 +126,21 @@ public UpdateControl<VirtualKafkaCluster> reconcile(VirtualKafkaCluster cluster,
         return uc;
     }
 
+    @NonNull
+    private static <R extends CustomResource<?, ?>> Stream<String> getMissingProxy(
+                                                                                   String prefix,
+                                                                                   Class<R> crdClass,
+                                                                                   TreeSet<? extends LocalRef<R>> refs) {
+        return refs.isEmpty() ? Stream.of()
+                : Stream.of(
+                        prefix + refs.stream()
+                                .map(ref -> ResourcesUtil.slug(crdClass, ref.getName()))
```

Comment:
I wonder if we could add a `getReferentType()` to LocalRef to get a hold of the CRD class without having to pass around two the class in addition to the ref. Alternatively we could use the strings from the localRef to build the slug like `lower(kind) + "." + group + "/" + name. But I can see having something based on the CR class is better if we need the same string and we're working with the resource rather than a ref.

Reply:
I agree this is annoying. The `LocalType` doesn't know the `singular` or `plural`, only the `kind`, because that's what the user specifies in the whatever resource's `spec`. We can't always define that statically (e.g. `AnyLocalType` won't know). That's also true for knowing a Class<T> (for some `T extends HasMetadata`): It works for the subtypes of `LocalRef` which have a statically known type due to what a CR defines. But we will eventually have a list of references which can support different `kinds`, modelled as `List<AnyLocalRef>`.